### PR TITLE
[FIX] if nocreate=1 then also see that as true

### PR DESCRIPTION
**Pull request by thomaspaulb**
Some people in their custom modules have used 'no_create=1' instead of True to disable create or edit, which [works](https://github.com/OCA/OCB/blob/10.0/addons/web/static/src/js/views/form_common.js#L235) in standard Odoo. But when `web_m2x_options` is installed, it will stop working, which is IMO undesired/unintended behavior.

### CODE REVIEW

File: web_m2x_options/static/src/js/form.js
Comment (hbrunn) [inline web_m2x_options/static/src/js/form.js:90]: please just pyeval the string, which is the reason it works in core

**Review by hbrunn — CHANGES_REQUESTED** [review]

File: web_m2x_options/static/src/js/form.js
Comment (hbrunn) [reply web_m2x_options/static/src/js/form.js:90]: ...pyeval or json?

Comment (thomaspaulb) [reply web_m2x_options/static/src/js/form.js:90]: [pyeval](https://github.com/OCA/OCB/blob/10.0/addons/web/static/src/js/views/form_common.js#L662) is used to convert the options dictionary into a Javascript object, after which a simple [falsy/truthy test](https://github.com/OCA/OCB/blob/10.0/addons/web/static/src/js/views/form_common.js#L235) is used to determine the value, which is why it works in core.

Since I am dealing with the `options` object that is already filled using `pyeval`, maybe the correct way is then just to replace the `is_option_set` check by a falsy/truthy test?

Comment (thomaspaulb) [reply web_m2x_options/static/src/js/form.js:90]: (not all `is_option_set` checks, but eg. [this one](https://github.com/OCA/web/blob/10.0/web_m2x_options/static/src/js/form.js#L400))

Comment (hbrunn) [reply web_m2x_options/static/src/js/form.js:90]: but I wonder how strings end up there in the first place? If somebody writes `<... options="{'test': True}" />`, this should come out as boolean, not string, right? This seems to me we do the parsing somewhere else wrongly. Do you agree? So we should pyeval when we read this?

Comment (hbrunn) [reply web_m2x_options/static/src/js/form.js:90]: it's not in the options attribute anyways as you write above which makes part of my reaction bullshit, but i stick to that we should look at parsing and not add stuff that happens to end up there as a string

**Review by hbrunn — APPROVED** [review]
why do we need the switch between types in the first place? I believe in some earlier versions we wanted to be nice for input, and then started that. But that can't be asked for a contribution to change, so I applaud fixing the symptom, this will be fine here. Thanks!

File: web_m2x_options/static/src/js/form.js
Comment (thomaspaulb) [reply web_m2x_options/static/src/js/form.js:90]: As an alternative I could do this: https://github.com/sunflowerit/web/pull/3. Do you have a preference?

**Review by NL66278 — APPROVED** [review]
:+1: LGTM

**Comment by gjotten** [comment]
@hbrunn

**Comment by etobella** [comment]
@thomaspaulb Can you check the issues with travis?

**Comment by thomaspaulb** [comment]
@simahawk Rebased

**Comment by thomaspaulb** [comment]
Okay I don't really know what to do here :/

**Comment by simahawk** [comment]
It's failing in the same way on 10.0.

image

@sbidoul do we still expect builds for 10.0 to work?

**Comment by sbidoul** [comment]
> do we still expect builds for 10.0 to work?

@simahawk dunno. This is Travis & MQT, it's been ages since I looked into that. So I guess the answer is no.

**Comment by thomaspaulb** [comment]
I've tried to use [guide](https://odoo-community.org/groups/contributors-15/contributors-253387) to change this to Github actions, but I'm running into too many issues to fix. Leaving this up to @OCA/web-maintainers to decide whether to manually force-merge this, skipping CI, or close.

**Comment by NL66278** [comment]
@thomaspaulb You can switch to github actions if you remove the pre-commit job from it. GitHub no longer supports python2. pre-commit can still be used locally to check before pushing to github.

**Comment by dreispt** [comment]
I'm doing a manual merge.

**Pull request merged**
